File: csspack.php
#!/usr/bin/env php
<?php

echo
	// remove unnecessary blanks
	preg_replace
	(
		'![ ]+\{!',
		'{',
		preg_replace
		(
			'!([:\(\{,;])[ ]+!',
			'\1',
			preg_replace
			(
				'![ ]+([\)\}])!',
				'\1',
				// condense blanks
				preg_replace
				(
					'![\t ]+!',
					' ',
					// remove newlines and white space at the beginning
					// and at the end of lines
					preg_replace
					(
						'![\t ]*[\r\n]+[\t ]*!m',
						'',
						// remove comments
						preg_replace
						(
							'!/\*[^*]*\*+([^/][^*]*\*+)*/!',
							'',
							file_get_contents( "php://stdin" )
						)
					)
				)
			)
		)
	) . "\n";

File: csspack.lua
#!/usr/bin/lua

local s = io.read( "*all" )

-- remove comments
s = s:gsub( "/[*]+.-[*]+/", "" )
-- remove newlines and white space at the beginning and at the end of lines
s = s:gsub( "[\t ]*[\r\n]+[\t ]*", "" )
-- condense tabs and blanks
s = s:gsub( "[\t ]+", " " )
-- remove unnecessary blanks
s = s:gsub( "[ ]+\{", "{" )
s = s:gsub( "([:\(\{,;])[ ]+", "%1" )
s = s:gsub( "[ ]+([\)\}])", "%1" )

io.write( s.."\n" )
